runtime-components/non-repudiation-app/src/main/scala/com/daml/nonrepudiation/app/OptionParser.scala

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.nonrepudiation.app
+
+import java.net.InetSocketAddress
+
+private[app] object OptionParser extends scopt.OptionParser[Configuration](NonRepudiationApp.Name) {
+
+  head(NonRepudiationApp.Name)
+
+  opt[String]("ledger-host")
+    .required()
+    .action(setLedgerHost)
+    .text("The host address of the participant. Required.")
+
+  opt[String]("ledger-port")
+    .required()
+    .action(setLedgerPort)
+    .text("The port of the participant. Required.")
+
+  opt[Map[String, String]]("jdbc")
```

Comment:
At some point we have to unify CLI parameters for this. At the moment, daml on sql, the JSON API, the trigger service and now the non-repudiation middleware all specify the database url slightly differently.

We should also open an issue to make this configurable via an environment variable (not because I think it’s really all that useful for security purposes but it’s often a policy for our customers to not specify any secrets via CLI).

Reply:
I agree, initially I had my own names, then I realized I was completely off compared to the rest of the runtime components.

Reply:
Tracked by https://github.com/digital-asset/daml/issues/9095